### ISSUE TITLE
JVM crash with exit code 134 when executing a query

### ISSUE THREAD

**Issue by paulbares** [duckdb/duckdb#10824]
### What happens?

Running a sql query leads to JVM crash with exit status code 134. The query involves a nested query and is using `array_agg` function. Details to reproduce below.

### To Reproduce

I wrote the simple test class to reproduce the issue. The faulty query is this one:
```sql
select "___alias___portfolio_Ticker___", "___alias___portfolio_Portfolio___", count(*) as "_contributors_count_", array_agg("___alias___DateScenario_any_value___") as "VaR 95_1", max("___alias___grouping_portfolio_Portfolio___") as "___grouping______alias___portfolio_Portfolio______", array_agg("___alias___ScenarioValue_sum___") as "VaR 95_0", max("___alias___grouping_portfolio_Ticker___") as "___grouping______alias___portfolio_Ticker______" from (select "portfolio"."Portfolio" as "___alias___portfolio_Portfolio___", "portfolio"."DateScenario" as "___alias___portfolio_DateScenario___", "portfolio"."Ticker" as "___alias___portfolio_Ticker___", grouping("portfolio"."Portfolio") as "___alias___grouping_portfolio_Portfolio___", grouping("portfolio"."Ticker") as "___alias___grouping_portfolio_Ticker___", any_value("portfolio"."DateScenario") as "___alias___DateScenario_any_value___", sum("portfolio"."ScenarioValue") as "___alias___ScenarioValue_sum___" from "portfolio" group by grouping sets(("___alias___portfolio_DateScenario___"), ("___alias___portfolio_Ticker___","___alias___portfolio_Portfolio___","___alias___portfolio_DateScenario___"), ("___alias___portfolio_Ticker___","___alias___portfolio_DateScenario___"))) group by "___alias___portfolio_Ticker___", "___alias___portfolio_Portfolio___" limit 10001
```

Test class:
```java
import org.duckdb.DuckDBConnection;

import java.io.IOException;
import java.io.InputStream;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.StandardCopyOption;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;

public class TestDuckDBQueryCrash {
  public static void main(String[] args) throws Exception {
    DuckDBConnection connection = createConnection();
    loadFile(connection, "portfolio.csv", "portfolio");

    try (Statement statement = connection.createStatement()) {
      ResultSet tableResult = statement.executeQuery("select * from portfolio limit 10");
      int c = 0;
      while (tableResult.next()) {
        c++;
      }
      System.out.println("First query, number of lines " + c);
    }
    String sql = """
            select "___alias___portfolio_Ticker___", "___alias___portfolio_Portfolio___", count(*) as "_contributors_count_", array_agg("___alias___DateScenario_any_value___") as "VaR 95_1", max("___alias___grouping_portfolio_Portfolio___") as "___grouping______alias___portfolio_Portfolio______", array_agg("___alias___ScenarioValue_sum___") as "VaR 95_0", max("___alias___grouping_portfolio_Ticker___") as "___grouping______alias___portfolio_Ticker______" from (select "portfolio"."Portfolio" as "___alias___portfolio_Portfolio___", "portfolio"."DateScenario" as "___alias___portfolio_DateScenario___", "portfolio"."Ticker" as "___alias___portfolio_Ticker___", grouping("portfolio"."Portfolio") as "___alias___grouping_portfolio_Portfolio___", grouping("portfolio"."Ticker") as "___alias___grouping_portfolio_Ticker___", any_value("portfolio"."DateScenario") as "___alias___DateScenario_any_value___", sum("portfolio"."ScenarioValue") as "___alias___ScenarioValue_sum___" from "portfolio" group by grouping sets(("___alias___portfolio_DateScenario___"), ("___alias___portfolio_Ticker___","___alias___portfolio_Portfolio___","___alias___portfolio_DateScenario___"), ("___alias___portfolio_Ticker___","___alias___portfolio_DateScenario___"))) group by "___alias___portfolio_Ticker___", "___alias___portfolio_Portfolio___" limit 10001
                        """;
    try (Statement statement = connection.createStatement()) {
      for (int i = 0; i < 10; i++) {
        ResultSet tableResult = statement.executeQuery(sql);
        int c = 0;
        while (tableResult.next()) {
          c++;
        }
        System.out.println("Second query, number of lines " + c);
      }
    }
  }

  private static DuckDBConnection createConnection() {
    try {
      Class.forName("org.duckdb.DuckDBDriver");
      return (DuckDBConnection) DriverManager.getConnection("jdbc:duckdb:");
    } catch (Exception e) {
      throw new RuntimeException(e);
    }
  }

  private static void loadFile(DuckDBConnection connection, String fileName, String table) {
    Path tempPath = null;
    try {
      InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream(fileName);
      tempPath = Files.createTempFile(fileName + "_tmp_" + System.currentTimeMillis(), ".csv");
      Files.copy(in, tempPath, StandardCopyOption.REPLACE_EXISTING); // copy the file in tmp dir. otherwise, issue can happen in docker container (file system does not exist)
      Statement statement = connection.createStatement();
      statement.execute("CREATE TABLE " + table + " AS SELECT * FROM read_csv_auto('" + tempPath + "');");
    } catch (Exception e) {
      throw new RuntimeException(e);
    } finally {
      if (tempPath != null) {
        try {
          Files.delete(tempPath);
        } catch (IOException e) {
          throw new RuntimeException(e);
        }
      }
    }
  }
}
```

The data is loaded from the csv file attached here.
[portfolio.csv](https://github.com/duckdb/duckdb/files/14386834/portfolio.csv)

Console output:
```
First query, number of lines 10

Process finished with exit code 134 (interrupted by signal 6:SIGABRT)
```

There is no exception raised nor core dump files after running this code. 

### OS:

MacOS Sonoma 14.2.1

### DuckDB Version:

0.10.0

### DuckDB Client:

Java

### Full Name:

Paul Bares

### Affiliation:

SquashQL

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a release build (and could not test with a nightly build)

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by szarnyasg** [comment]
Thanks, I could indeed reproduce the issue.

**Comment by paulbares** [comment]
@szarnyasg is there anything I can do on my side to avoid this issue?

**Comment by szarnyasg** [comment]
@paulbares I'm not aware of a simple workaround. I ran the same query in the CLI and Python clients, where it ran fine, so the issue seems to be related to some of DuckDB's Java bindings. Breaking down the query to smaller parts may help but is not a guarantee for success.

**Comment by paulbares** [comment]
I simplified the query. The problem can be reproduced with this query:
```sql
SELECT array_agg(datescenario)
FROM   (SELECT portfolio,
               datescenario,
               ticker
        FROM   portfolio
        GROUP  BY ticker,
                  portfolio,
                  datescenario)  
```

**Comment by paulbares** [comment]
Any updates on this issue?

**Comment by paulbares** [comment]
This issue seems fixed in the 0.10.1 release (I can't reproduce it anymore). Maybe one of those PRs contains the fix:
- https://github.com/duckdb/duckdb/pull/11161
- https://github.com/duckdb/duckdb/pull/10911
